spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/web/trace/reactive/ServerWebExchangeTraceableRequest.java

```diff
@@ -45,8 +48,9 @@
 		this.method = request.getMethodValue();
 		this.headers = request.getHeaders();
 		this.uri = request.getURI();
-		this.remoteAddress = (request.getRemoteAddress() != null)
-				? request.getRemoteAddress().getAddress().toString() : null;
+		this.remoteAddress = Optional.ofNullable(request.getRemoteAddress())
```

Comment:
We prefer not to use `Optional` for internal `null` checks such as this. If the readability of the ternary is a problem it could be pulled out into a method that's something like this instead:

```java
private static String asString(InetSocketAddress socketAddress) {
	return (socketAddress != null && socketAddress.getAddress() != null)
			? socketAddress.getAddress().toString() : null;
}
```